internal/controller/persistentvolumeclaim_controller.go

```diff
@@ -66,16 +58,81 @@ func (r *PersistentVolumeClaimReconciler) Reconcile(ctx context.Context, req ctr
 		}, nil
 	}
 
-	// Skip if the PVC is not deleted or PVC does not have TopoLVM's finalizer.
-	if pvc.DeletionTimestamp == nil {
+	// Skip if PVC does not have TopoLVM's finalizer.
+	if !controllerutil.ContainsFinalizer(pvc, topolvm.PVCFinalizer) {
 		return ctrl.Result{}, nil
 	}
 
-	if !controllerutil.ContainsFinalizer(pvc, topolvm.PVCFinalizer) {
-		return ctrl.Result{}, nil
+	if pvc.DeletionTimestamp == nil {
+		err := r.notifyKubeletToResizeFS(ctx, log, pvc)
+		return ctrl.Result{}, err
+	}
+
+	result, err := r.deletePodsUsingDeletingPVC(ctx, log, pvc)
+	if result.Requeue || err != nil {
+		return result, err
+	}
+
+	controllerutil.RemoveFinalizer(pvc, topolvm.PVCFinalizer)
```

Comment:
```suggestion
	updated := controllerutil.RemoveFinalizer(pvc, topolvm.PVCFinalizer)
```
you can use updated to conditionally remove the finalizer, should (almost) never happen, but still good practice

Reply:
I found we checked the PVC has a finalizer on line 62, the condition must be true. I leave it as before.